edx-solutions/edx-platform
# Add endpoint to get SAML providers for a user.

Description goes here. e.g. This PR contains the LibraryContent XBlock, which allows to display library content in a course.

**JIRA tickets**: Implements OC-4285, MCKIN-7245

**Discussions**: Design is being discussed on https://docs.google.com/document/d/115PkSUHpxegaalwQzzDSAV7SWqUNljPrI18iF2udfdw/edit#

**Dependencies**: None

**Screenshots**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None

**Testing instructions**:

1. As an unauthenticated user, access /api/third_party_auth/v0/users/<username>/providers/ for a user with at least one third-party-auth connection.  Verify that the providers are included in the response.
2. Replace <username> with the same user's email address, and verify that the same response is returned.
3. Replace <username> with the username or email address of a user that doesn't have any third-party-auth connections.  Verify the the returned list of providers is empty.
4. Replace <username> with a nonexistent username.  Verify that a 200 response is returned with an empty list of providers.
5. Run third-party-auth unittests: `paver test_system -s lms -t common/djangoapps/third_party_auth/api/tests/`
6. Hit the api several times (11) in one minute.  Verify that a 429 (throttled) response is eventually returned.  If you're feeling ambitious, send 51 requests over the course of 10 minutes, and verify that the daily throttle limit gets hit as well.


**Author notes and concerns**:

* ~~Throttling seems to be happening, but not according to the configuration I provided.  Tested via curl on the command line.~~ Throttling works now.
* Tests are failing due to NPM installation issues.  I spent a few hours trying to resolve the problem this morning, with no luck.  I'll ask around if anyone else has faced this, and if not I'll try to deal with it later.  In the meantime, please review with manual testing, and also run unit
**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```

Here's the PR.  I'll fill out details in the morning.  



Reviewers:
